canonical-web-and-design/maas-ui
# Add GA events to KVM details + store NUMA view toggle state in local storage

## Done

- Added GA events to KVM details (NUMA view toggle, "Show more/less storage pools", "Show more/less NUMA nodes")
- Stored NUMA view toggle state in local storage, on a per-KVM basis

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Load a KVM details page and check that the aggregate view shows by default.
- Toggle to NUMA node view and refresh that page.
- Check that the page is still set to NUMA node view.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2080
Fixes canonical-web-and-design/MAAS-squad#2086
